test/e2e/apps/cronjob.go

```diff
@@ -456,7 +470,9 @@ var _ = SIGDescribe("CronJob", func() {
 		cjs, err = cjClient.List(context.TODO(), metav1.ListOptions{LabelSelector: "special-label=" + f.UniqueName})
 		framework.ExpectNoError(err)
 		// Should have <= 2 items since some cronjobs might not have been deleted yet due to finalizers
-		framework.ExpectEqual(len(cjs.Items) <= 2, true, "filtered list should be <= 2")
+		if len(cjs.Items) > 2 {
+			framework.Fail("filtered list should be <= 2")
```

Comment:
Based on the discussion in https://kubernetes.slack.com/archives/C09QZ4DQB/p1654236320527119, it may be better to log the list before failing:

```suggestion
			framework.Logf("got unexpected filtered list: %v", cjs.Items)
			framework.Fail("filtered list should be <= 2")
```